authenticate-oauth/Web/Authenticate/OAuth.hs

```diff
@@ -355,12 +375,21 @@ injectOAuthToCred oa cred =
             ] cred
 
 
+-- Note that prefix is used for realm in addAuthHeader, so should fix that up,
+-- but may need an API break for that.
+addAuthBody :: BS.ByteString -> Credential -> Request -> Request
+addAuthBody _ (Credential cred) req = urlEncodedBody (filterCreds cred) req
```

Comment:
1 is easy, will do. I'm not sure what 2 entails, is that that ^ - stuff ?

I've a revised patch that has the getAccessTokenWith as requested. I'm also adding in a change to the return type there to return an Either, which is useful for handling some errors (e.g. it enables polling). Should have it all pushed up in an hour or so.